### PR TITLE
[CMake] SofaMacros.cmake: deprecating sofa_create_package

### PR DESCRIPTION
**Pull request by marques-bruno**
In favor of a new sofa_generate_package function (with named arguments)

It's never clear what the arguments passed to this function are for. Now you have to know to use it :)

______________________________________________________

This PR: 
- [ ] builds with SUCCESS for all platforms on the CI.
- [ ] does not generate new warnings.
- [ ] does not generate new unit test failures.
- [ ] does not generate new scene test failures.
- [ ] does not break API compatibility.
- [ ] is more than 1 week old (or has fast-merge label).

**Reviewers will merge only if all these checks are true.**

### CODE REVIEW

**Comment by damienmarchal** [comment]
I fully agree with the systematic use of name argument in CMake macros.

File: SofaKernel/extlibs/newmat/CMakeLists.txt [hunk 1]
```diff
@@ -56,4 +56,4 @@ endif()
 
 
 include(../../SofaFramework/SofaMacros.cmake)
-sofa_create_package(Newmat 0 ${PROJECT_NAME} ${PROJECT_NAME})
+sofa_generate_package(NAME Newmat VERSION 0 TARGETS ${PROJECT_NAME} INCLUDES ${PROJECT_NAME})
```
Comment (epernod) [inline SofaKernel/extlibs/newmat/CMakeLists.txt]: Do you really need to pass the argument parameter names: "NAME, VERSION, ..." ?
In the cmake function I use I directly give the parameter values.

Comment (marques-bruno) [reply SofaKernel/extlibs/newmat/CMakeLists.txt]: This is the whole point of the PR actually :)
Unnamed arguments are ambiguous, as nobody ever really knows what they are for, without taking a look at the macros themselves, which nearly no one does.

**Review by damienmarchal — APPROVED** [review]
I think this is a good idea to only have named arguments.

File: SofaKernel/extlibs/newmat/CMakeLists.txt (hunk 1 above)
Comment (epernod) [reply SofaKernel/extlibs/newmat/CMakeLists.txt]: ok

Comment (damienmarchal) [reply SofaKernel/extlibs/newmat/CMakeLists.txt]: INCLUDES_ROOT_DIR

**Comment by marques-bruno** [comment]
[with-scene-tests][with-regression-tests]

**Comment by marques-bruno** [comment]
[ci-build][with-scene-tests][with-regression-tests]

**Pull request merged**